sources/ascwp/src/main.rs

```diff
@@ -0,0 +1,59 @@
+mod config;
+mod error;
+use crate::config::Config;
+use crate::error::Result;
+use imdsclient::ImdsClient;
+use snafu::{OptionExt, ResultExt};
+use std::fs;
+use log::{error, info, warn};
+use std::time::Duration;
+use async_std::task;
+
+
+
+//Path to config file containing host's eks_autoscaling.should-wait setting
+const CONFIG_PATH: &str = "/etc/ascwp.toml";
+//Marker file that is created after first run. We only want this program to run once.
+const MARKER_FILE: &str = "/var/lib/bottlerocket/ascwp.ran";
+
+//Uses an imdsclient function to fetch the lifecyclestate of the host
+async fn get_lifecycle_state(client: &mut ImdsClient) -> Result<String> {
+    client
+        .fetch_lifecycle_state()
+        .await
+        .context(error::ImdsRequestSnafu)?
+        .context(error::ImdsNoneSnafu {
+            what: "instance-id",
+        })
+}
+
+//Continously fetches lifecycle state of the host. Returns when host is "InService"
+async fn wait_until_inservice() -> Result<()> {
+    let mut client = ImdsClient::new();
+    let mut lifecycle_state = get_lifecycle_state(&mut client).await?;
+    while lifecycle_state.ne("InService") {
+        lifecycle_state = get_lifecycle_state(&mut client).await?;
+        task::sleep(Duration::from_secs(5)).await;
```

Comment:
I believe we typically use `tokio`'s [sleep implementation](https://docs.rs/tokio/latest/tokio/time/fn.sleep.html). You may be able to remove an additional dependency that way.

Separately: The poll interval should be hoisted to a `CONSTANT`, or perhaps even moved to the config file (with a default value).